x-pack/filebeat/module/crowdstrike/falcon/config/pipeline.js

```diff
@@ -0,0 +1,132 @@
+        fields: ["message"],
+        target: "crowdstrike",
+        process_array: true,
+        max_depth: 8
+    });
+
+    var dropFields = function(evt) {
+        evt.Delete("message");
+        evt.Delete("host.name");
+    };
+  
+    var setFields = function (evt) {
+        evt.Put("agent.name", "falcon");
+    };
+
+    var convertFields = new processor.Convert({
+        fields: [
+            // DetectionSummaryEvent
+            { from: "crowdstrike.event.LocalIP", to: "source.ip" },
+            { from: "crowdstrike.event.ProcessId", to: "process.pid" },
+
+            // UserActivityAuditEvent and AuthActivityAuditEvent
+            { from: "crowdstrike.event.UserIp", to: "source.ip" },
+        ],
+        mode: "copy",
+        ignore_missing: true,
+    });
+
+    var parseTimestamp = new processor.Timestamp({
+        field: "crowdstrike.metadata.eventCreationTime",
+        target_field: "@timestamp",
+        timezone: "UTC",
+        layouts: ["UNIX_MS"],
+        ignore_missing: false,
+    });
+
+    var processEvent = function(evt) {
+        var eventType = evt.Get("crowdstrike.metadata.eventType")
+
+        evt.Put("agent.type", "falcon")
+        evt.Put("event.kind", "event")
+        evt.Put("host.name", "")
+
+        switch (eventType) {
+            case "DetectionSummaryEvent":
+                var tactic = evt.Get("crowdstrike.event.Tactic").toLowerCase()
+                var technique = evt.Get("crowdstrike.event.Technique").toLowerCase()
+                  
+                evt.Put("event.action", tactic + "_" + technique)
+                evt.Put("event.kind", "alert")
+                evt.Put("event.type", "info")
+                evt.Put("event.category", "malware")
+                evt.Put("event.severity", evt.Get("crowdstrike.event.Severity"))
+                evt.Put("message", evt.Get("crowdstrike.event.DetectDescription"))
+                evt.Put("process.name", evt.Get("crowdstrike.event.FileName"))
+                evt.Put("process.executable", evt.Get("crowdstrike.event.CommandLine"))
+                evt.Put("user.name", evt.Get("crowdstrike.event.UserName"))
+                evt.Put("user.domain", evt.Get("crowdstrike.event.MachineDomain"))
+                evt.Put("rule.reference", evt.Get("crowdstrike.event.FalconHostLink"))
+                evt.Put("agent.id", evt.Get("crowdstrike.event.SensorId"))
+                evt.Put("host.name", evt.Get("crowdstrike.event.ComputerName"))
+
+                break;
+
+            case "UserActivityAuditEvent":
+                evt.Put("user.name", evt.Get("crowdstrike.event.UserId"))
+                evt.Put("message", evt.Get("crowdstrike.event.OperationName"))
+                evt.Put("event.action", convertUnderscore(eventType))
+                evt.Put("event.type", "change")
+
+                break;
+
+            case "AuthActivityAuditEvent":
+                evt.Put("user.name", evt.Get("crowdstrike.event.UserId"))
```

Comment:
Since Crowdstrike seems to alternate between providing a only `UserId` or only a `UserName`, I'd suggest to be strict in populating

* `user.id` with `UserId`
* `user.name` with `UserName`

Then if we want to conditionally backfill `user.name` with `UserId` when `user.name` is not present, that may make sense.

I wonder if there's a relation with their API service logs always filling `UserId` and the endpoint events filling `UserName`?